pandas/tests/io/parser/common/test_chunksize.py

```diff
@@ -260,8 +257,12 @@ def test_warn_if_chunks_have_mismatched_type(all_parsers):
             "Specify dtype option on import or set low_memory=False.",
             buf,
         )
-
-    assert df.a.dtype == object
+    if parser.engine == "c" and parser.low_memory:
```

Comment:
Shouldn't `low_memory` still be using the proper data type? Or why would that stick to object?

Reply:
I am not super familiar with the parser code, but I _think_ that with the low memory parser, parsing is done in chunks, and so if the inference changes later on, you end up with chunks with different types, and then get object dtype as a result.

In the test here, we have a column with mostly integers, and only a few strings in the middle. So with the default parser, it will decide based on the values in the full column that the dtype should be string. But chunk by chunk you get some chunks as integer and some as string

Reply:
Ah I see...yea that's a weird one